archive/tar_stream.ts

```diff
@@ -0,0 +1,450 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * The interface required to provide a file.
+ */
+export interface TarStreamFile {
+  /**
+   * The path to the file, relative to the archive's root directory.
+   */
+  path: string;
+  /**
+   * The size of the file in bytes.
+   */
+  size: number;
+  /**
+   * The contents of the file.
+   */
+  readable: ReadableStream<Uint8Array>;
+  /**
+   * The metadata of the file.
+   */
+  options?: TarStreamOptions;
+}
+
+/**
+ * The interface required to provide a directory.
+ */
+export interface TarStreamDir {
+  /**
+   * The path of the directory, relative to the archive's root directory.
+   */
+  path: string;
+  /**
+   * The metadata of the directory.
+   */
+  options?: TarStreamOptions;
+}
```

Comment:
@crowlKats, did you still want to add a `type` field to these interfaces?

Reply:
yes, to easily distinguish in code what entry is what kidn